### ISSUE TITLE
[Beta] Unable to scroll challenge description with keyboard up and down keys

### ISSUE THREAD

**Issue by rickysyamsi** [freeCodeCamp/freeCodeCamp#37158]
**Describe the bug**
I can not scroll challenge description with keyboard navigation (up and down keys).

**To Reproduce**
1. Go to any code challenge page (for example: http://localhost:8000/learn/responsive-web-design/basic-html-and-html5/say-hello-to-html-elements)
2. Click on challenge description (left part).
3. Try to scroll description with keyboard up and down keys.
4. It is not working.

**Expected behavior**
I should be able to scroll challenge description with keyboard navigation (up and down keys).

**Desktop (please complete the following information):**
 - OS: Windows 10 1809 Build 17763.805
 - Browser Chrome
 - Version 77.0.3865.90 (Official Build) (64-bit)

**Comment by rickysyamsi** [comment]
> Currently when using keyboard navigation, students need to press multiple tabs to access editor. I think that it will be very helpful if students can access editor easily with single keyboard press (for example 'e' key).

following suggestion by @ojeytonwilliams to discuss adding this as a feature.

**Comment by scissorsneedfoodtoo** [comment]
@rickysyamsi, thank you for catching this. 

Just checked on https://learn.freecodecamp.org/responsive-web-design/basic-html-and-html5/say-hello-to-html-elements/ and it's possible to scroll or navigate through the description with the arrow and tab keys. Tried disabling the recently added `<HotKeys>` components and playing with the overflow on some of the divs, but the issue remains when running locally.

Did more testing and it seems like adding the `tabindex` attribute and setting it to -1 or 0 on [this line](https://github.com/freeCodeCamp/freeCodeCamp/blob/f1596458108b8dce1dd2743178f2d177dcc826e8/client/src/templates/Challenges/components/Side-Panel.js#L68) allows for keyboard navigation in description again. However I'm not sure what the underlying issue is, and if setting `tabindex` here will cause problems with accessibility.

**Comment by ojeytonwilliams** [comment]
Hey @rickysyamsi, I followed up on your comments in the contributor chat and noticed that the elements that let you scroll (i.e. you click on them and the arrow keys work) are focusable. I wasn't sure why this mattered until found [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex):

> If you set the tabindex attribute on a `<div>`, then its child content cannot be scrolled with the arrow keys unless you set tabindex on the content, too.

So, the solution seems to be just to add `tabindex="-1"` to the instructions.

@scissorsneedfoodtoo apparently we were looking at the exact same thing :smile:.  Re: accessibility `tabindex="0"` would *probably* be a mistake, since it's not something that can be interacted with.  I'll read up on this to be sure.

**Comment by rickysyamsi** [comment]
@scissorsneedfoodtoo @ojeytonwilliams Hey guys, thank you for responding.

I had solved the problem with the same `tabindex="-1"` approach on my local environment, but I have not made pull request because I'm still experimenting with the feature possibility of re-focusing on editor from navigation mode easily with single keyboard press (for example 'e' key), instead of multiple tabs.

From my quick research, there should be no side effects because `tabindex="-1"` only makes an element programatically focusable and will not be reachable by sequential keyboard navigation (tab key).

Now regarding my little experiment, I think it is better if that feature is discussed more thoroughly in another issue. To be able to set focus on editor with react-hotkeys, the `<HotKeys>` parent component need to access ref to `<Editor>` child component.

It can be done with [Forwarding Refs](https://reactjs.org/docs/forwarding-refs.html), but the official docs specifically mention to be careful when using it [https://reactjs.org/docs/forwarding-refs.html#note-for-component-library-maintainers](https://reactjs.org/docs/forwarding-refs.html#note-for-component-library-maintainers).

Hence my suggestion since this is only supposed to be a nice to have feature and multiple tabs press is still working. Please make another issue if this feature is still desirable, I will be glad to dig on it.

Have a good day  :)

**Comment by ojeytonwilliams** [comment]
@rickysyamsi, you're right, this should have been a separate issue from the start.  My mistake - I'll create one now.
